sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -863,7 +864,12 @@ case class ShowTablesCommand(
       // Show the information of tables.
       val tables =
         tableIdentifierPattern.map(catalog.listTables(db, _)).getOrElse(catalog.listTables(db))
-      tables.map { tableIdent =>
+      val filterCachedTables = if (showCached) {
+        tables.filter(table => sparkSession.catalog.isCached(table.quotedString))
```

Comment:
I'm not sure if `isCached` can work nicely with streaming tables. Can we test that?

Reply:
IMO we can only cache batch table, with streaming we are allowed cache tables in `foreachBatch`. Isn't it?

Reply:
yeah I think we cannot cache streaming table, but here the `isCached` is called irrespective of whether the table is cached or not, is that correct?

Reply:
I think `SparkSession.table` inside the `isCached` method will create a `UnresolvedRelation` with `isStreaming = false`, and later on in the analysis this flag may cause failure when a streaming table is encountered with that flag.

Reply:
Based on current implementation of cache, we only cache the `LogicalPlan` without any metadata like `table name`. So here we list and filter the table one by one.

Reply:
> I think SparkSession.table inside the isCached method will create a UnresolvedRelation with isStreaming = false, and later on in the analysis this flag may cause failure when a streaming table is encountered with that flag.

I guess you consider about something like this ?
```

def lookupTempView(
    identifier: Seq[String], isStreaming: Boolean = false): Option[LogicalPlan] = {
  // Permanent View can't refer to temp views, no need to lookup at all.
  if (isResolvingView) return None

  val tmpView = identifier match {
    case Seq(part1) => v1SessionCatalog.lookupTempView(part1)
    case Seq(part1, part2) => v1SessionCatalog.lookupGlobalTempView(part1, part2)
    case _ => None
  }

  if (isStreaming && tmpView.nonEmpty && !tmpView.get.isStreaming) {
    throw new AnalysisException(s"${identifier.quoted} is not a temp view of streaming " +
      s"logical plan, please use batch API such as `DataFrameReader.table` to read it.")
  }
  tmpView
}
```

Reply:
Yes this looks like it. I encountered something similar in #30211 and not sure the same will cause issue for this PR as well.

Reply:
Thanks for your sharing!